src/core/tests/type_prop/squeeze.cpp

```diff
@@ -4,187 +4,266 @@
+    auto param = make_shared<op::Parameter>(element::f32, Shape{1, 2, 3, 4});
+    auto axes_node = make_shared<op::Constant>(element::i32, Shape{2, 1}, vector<int32_t>{0, 2});
 
-    ASSERT_EQ(squeeze_default_axes->get_element_type(), element::f32);
-    ASSERT_EQ(squeeze_default_axes->get_shape(), (Shape{4, 4, 8}));
+    OV_EXPECT_THROW(auto s = make_shared<op::Squeeze>(param, axes_node),
+                    NodeValidationFailure,
+                    HasSubstr("Second input (axes) should not be of rank higher than 1."));
 }
 
-TEST(type_prop, squeeze_unsqueezable_no_axes) {
-    auto param = make_shared<op::Parameter>(element::f32, PartialShape{Dimension(2, 5), Dimension(3, 4), 6});
-    auto squeeze = make_shared<op::Squeeze>(param);
+TEST(type_prop, squeeze_incorrect_negative_axes) {
+    auto param = make_shared<op::Parameter>(element::f32, Shape{1, 4, 1, 4, 1, 8});
+    auto axes_node = make_shared<op::Constant>(element::i64, Shape{2}, vector<int64_t>{-6, -10});
 
-    ASSERT_EQ(squeeze->get_element_type(), element::f32);
-    EXPECT_TRUE(squeeze->get_output_partial_shape(0).same_scheme(PartialShape{Dimension(2, 5), Dimension(3, 4), 6}));
+    OV_EXPECT_THROW(auto s = make_shared<op::Squeeze>(param, axes_node),
+                    ov::Exception,
+                    HasSubstr("Parameter axis -10 out of the tensor rank range"));
 }
 
-TEST(type_prop, squeeze_no_axes) {
-    auto param = make_shared<op::Parameter>(element::f32, Shape{1, 4, 1, 4, 1, 8});
-    auto squeeze = make_shared<op::Squeeze>(param);
+using TypePropTestParam = std::tuple<PartialShape, std::vector<int64_t>, PartialShape>;
 
-    ASSERT_EQ(squeeze->get_element_type(), element::f32);
-    ASSERT_EQ(squeeze->get_shape(), (Shape{4, 4, 8}));
+class SqueezeTest : public WithParamInterface<TypePropTestParam>, public UnSqueezeFixture {
+protected:
+    void SetUp() override {
+        std::tie(p_shape, axes, exp_shape) = GetParam();
+        UnSqueezeFixture::SetUp();
+    }
 
-    auto axes_node = make_shared<ngraph::op::Constant>(element::u64, Shape{0}, vector<int64_t>{});
-    auto squeeze_default_axes = make_shared<op::Squeeze>(param, axes_node);
+    std::pair<std::vector<size_t>, std::vector<size_t>> make_in_exp_labels() const {
+        std::vector<size_t> in_labels;
+        std::generate_n(std::back_inserter(in_labels), p_shape.size(), ov::SeqGen<size_t>(1));
+
+        std::set<int64_t> axes_to_remove;
+        if (axes.empty()) {
+            for (auto dim = p_shape.begin(); dim != p_shape.end(); ++dim) {
+                if (*dim == 1 || exp_shape.rank().is_dynamic()) {
+                    axes_to_remove.insert(std::distance(p_shape.begin(), dim));
+                }
+            }
+        } else {
+            for (const auto& axis : axes) {
+                axes_to_remove.insert(axis < 0 ? axis + p_shape.size() : axis);
+            }
+        }
+
+        auto rm_iter = axes_to_remove.begin();
+        size_t rm_idx = 0;
+        auto exp_labels = in_labels;
```

Comment:
How the `default_label_evaluator` works for Squeeze when the label of dimension to remove is not `1`?

Reply:
OK, But generally default label evaluator doesn't care about label value except `no_value`.